hazelcast/sql.py

```diff
@@ -0,0 +1,1604 @@
+            return error
+
+        return HazelcastSqlError(
+            self.get_client_id(), _SqlErrorCode.GENERIC, try_to_get_error_message(error), error
+        )
+
+    def close(self, connection, query_id):
+        """Closes the remote query cursor.
+
+        Args:
+            connection (hazelcast.connection.Connection): Connection
+                that the first execute request, hence the close request
+                must route to.
+            query_id (_SqlQueryId): The query id to close.
+
+        Returns:
+            Future:
+        """
+        request = sql_close_codec.encode_request(query_id)
+        invocation = Invocation(request, connection=connection)
+        self._invocation_service.invoke(invocation)
+        return invocation.future
+
+
+class SqlExpectedResultType(object):
+    """The expected statement result type."""
+
+    ANY = 0
+    """
+    The statement may produce either rows or an update count.
+    """
+
+    ROWS = 1
+    """
+    The statement must produce rows. An exception is thrown is the statement produces an update count.
+    """
+
+    UPDATE_COUNT = 2
+    """
+    The statement must produce an update count. An exception is thrown is the statement produces rows.
+    """
+
+
+class SqlStatement(object):
+    """Definition of an SQL statement.
+
+    This object is mutable. Properties are read once before the execution
+    is started. Changes to properties do not affect the behavior of already
+    running statements.
+    """
+
+    TIMEOUT_NOT_SET = -1
+
+    TIMEOUT_DISABLED = 0
+
+    DEFAULT_TIMEOUT = TIMEOUT_NOT_SET
+
+    DEFAULT_CURSOR_BUFFER_SIZE = 4096
+
+    def __init__(self, sql):
+        self.sql = sql
+        self._parameters = []
+        self._timeout = SqlStatement.DEFAULT_TIMEOUT
+        self._cursor_buffer_size = SqlStatement.DEFAULT_CURSOR_BUFFER_SIZE
+        self._schema = None
+        self._expected_result_type = SqlExpectedResultType.ANY
+
+    @property
+    def sql(self):
+        """str: The SQL string to be executed."""
+        return self._sql
+
+    @sql.setter
+    def sql(self, sql):
+        check_not_none(sql, "SQL cannot be None")
+
+        if not sql.strip():
+            raise ValueError("SQL cannot be empty")
+
+        self._sql = sql
+
+    @property
+    def schema(self):
+        """str: The schema name. The engine will try to resolve the
+        non-qualified object identifiers from the statement in the given
+        schema. If not found, the default search path will be used, which
+        looks for objects in the predefined schemas ``partitioned`` and
+        ``public``.
+
+        The schema name is case sensitive. For example, ``foo`` and ``Foo``
+        are different schemas.
+
+        The default value is ``None`` meaning only the default search path is
+        used.
+        """
+        return self._schema
+
+    @schema.setter
+    def schema(self, schema):
+        self._schema = schema
```

Comment:
Would a regular `schema` class variable work better? Since the getter and setter don't seem to do anything special.

Reply:
If it is okay, I would like to leave it like this. It makes documentation more clear and it looks the same as other properties. Otherwise, this would be shown differently.

See the picture below. Not a big deal, but I would like to be consistent.

![Screenshot from 2021-06-01 16-39-52](https://user-images.githubusercontent.com/29387993/120333225-25653600-c2f8-11eb-880d-68efaa73d472.png)